core/src/main/java/io/hyperfoil/core/generators/RandomStringStep.java

```diff
@@ -0,0 +1,102 @@
+import io.hyperfoil.api.config.Name;
+import io.hyperfoil.api.config.Step;
+import io.hyperfoil.api.config.StepBuilder;
+import io.hyperfoil.api.session.Access;
+import io.hyperfoil.api.session.ResourceUtilizer;
+import io.hyperfoil.api.session.Session;
+import io.hyperfoil.core.builders.BaseStepBuilder;
+import io.hyperfoil.core.session.SessionFactory;
+
+public class RandomStringStep implements Step, ResourceUtilizer {
+   private final Access toVar;
+   private final StringGeneratorType stringGeneratorType;
+
+   public RandomStringStep(Access toVar, StringGeneratorType stringGeneratorType) {
+      this.toVar = toVar;
+      this.stringGeneratorType = stringGeneratorType;
+   }
+
+   @Override
+   public boolean invoke(Session session) {
+      toVar.setObject(session, this.stringGeneratorType.generate());
+      return true;
+   }
+
+   @Override
+   public void reserve(Session session) {
+      toVar.declareObject(session);
+   }
+
+   /**
+    * Stores random string into session variable based on the generator.
+    */
+   @MetaInfServices(StepBuilder.class)
+   @Name("randomString")
+   public static class Builder extends BaseStepBuilder<Builder> implements InitFromParam<Builder> {
+      private String toVar;
+      private StringGeneratorType stringGeneratorType;
+
+      @Override
+      public Builder init(String rangeToVar) {
+         if (rangeToVar == null) {
+            return this;
+         }
+         String arrow = "<-";
+         int arrowIndex = rangeToVar.indexOf(arrow);
+         String generator = rangeToVar.substring(arrowIndex + arrow.length()).trim();
+         if (arrowIndex < 0 || generator.isEmpty()) {
+            throw new BenchmarkDefinitionException("Expecting format var <- generator");
+         }
+         try {
+            stringGeneratorType = StringGeneratorType.valueOf(generator.toUpperCase());
+         } catch (IllegalArgumentException e) {
+            throw new BenchmarkDefinitionException(String.format("generator %s not found. Supported are: %s",
+                  generator, Arrays.toString(StringGeneratorType.values()).toLowerCase()));
+         }
+         toVar = rangeToVar.substring(0, arrowIndex).trim();
+         return this;
+      }
+
+      /**
+       * Variable name to store the result.
+       *
+       * @param var Variable name.
+       * @return Self.
+       */
+      public Builder toVar(String var) {
+         this.toVar = var;
+         return this;
+      }
+
+      @Override
+      public List<Step> build() {
+         if (toVar == null || toVar.isEmpty()) {
+            throw new BenchmarkDefinitionException("Missing target var.");
+         }
+         if (stringGeneratorType == null) {
+            throw new BenchmarkDefinitionException("Missing target generator.");
+         }
+         return Collections.singletonList(new RandomStringStep(SessionFactory.access(toVar), stringGeneratorType));
+      }
+   }
+
+   private enum StringGeneratorType {
```

Comment:
What's the advantage in having
```
- randomStringGenerator: foo <- UUID
- randomStringGenerator: bar <- SomeOtherEnumValue
```
vs.
```
- randomUUID: foo
- randomSomethingElse: bar
```
The second variant seems more concise and can use code completion (IDE can suggest you randomUUID or randomSomethingElse, it would have hard time suggesting you enum values in a custom string).

Reply:
Renamed to RandomUUIDStep